### PR TITLE
Issue 223/Created DeleteClientModal

### LINKED ISSUE #223: Make the alert popup on delete of client a MUI modal component (ClientListTableRow.jsx)
### Issue:

When clicking the `delete` icon in the table of clients, an alert message pops up to confirm.  Need to use an MUI modal and styling to match rest of app instead.

---

### Direction:

Take inspiration from our other MUI modals to have similar styling for this. 
`LogoutModal`, `AddClientModal` (which isn't 100% complete yet), and even `InactivityMessage` are all closely styled and fit within the app theme.

Screenshot 2023-06-06 at 2 58 12 PM

### PR DESCRIPTION
**Pull request by ajay2827**
Created DeleteClientModal

### CODE REVIEW

**Comment by xscottxbrownx** [comment]
@ajay2827  one more thing (all good for first time to have issues)...

When creating your PR make sure to choose the `Development` branch for which branch to merge your changes into. We do not push them straight to the `Master` branch.

Screenshot 2023-06-16 at 7 26 06 AM
Screenshot 2023-06-16 at 7 26 15 AM


I already changed it for this PR (as you can see above)

**Comment by xscottxbrownx** [comment]
@ajay2827 I don't see your branch in this repo...

did you fork this repo instead of clone?

File: src/components/Clients/ClientList.jsx [hunk 1]
```diff
@@ -35,7 +35,7 @@ const ClientList = () => {
         }}
       >
         <Typography variant="h6" component="h2" mb={2} align="center" color="secondary">
-          Add clients to your list
```
Comment (xscottxbrownx) [inline src/components/Clients/ClientList.jsx:38]: **1)** There shouldn't be any changes to this file in this PR that is scoped to solve the issue with the delete client modal. Has nothing to do with the Issue/PR at hand.


**2)** Why is this being deleted?

This is what shows in the `Clients` page if the user doesn't have a client list yet.

**Comment by ajay2827** [comment]
> @ajay2827 I don't see your branch in this repo...
> 
> did you fork this repo instead of clone?

By mistake first, i clone it, now i realize it is instead of cloning it i should fork it

File: src/components/DeleteModal/DeleteModal.jsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+      >
+        NO
+      </Button>
+
+        <Button
+          variant="outlined"
+          color="success"
+          endIcon={<CheckIcon />}
+          sx={{ marginLeft: '1rem' }}
+          onClick={handleDelete}
```
Comment (xscottxbrownx) [inline src/components/DeleteModal/DeleteModal.jsx]: This `handleDelete` which is really `handleDeleteClient`....

**1)** I personally would prefer the naming stays detailed (part of self-documenting code, "what are we deleting? a client").    When working with a larger team, especially remotely that doesn't meetup as often, we want to keep the code specific, so every dev who encounters the codebase easily understands what exactly is happening.

**2)** This onClick function doesn't change the `showDeleteModal` state, so when a user clicked this button to approve deleting a client the modal will not go away - which is unexpected behavior and makes the user have to click a different button afterwords or click outside of the modal.

File: src/components/NavBar/NavBar.jsx [hunk 3]
```diff
@@ -16,7 +16,7 @@ import Toolbar from '@mui/material/Toolbar';
 import Typography from '@mui/material/Typography';
 import { useTheme } from '@mui/material/styles';
 // Component Imports
-import LogoutModal from '../LogoutModal/LogoutModal';
+import Modal from '../LogoutModal/LogoutModal';
```
Comment (xscottxbrownx) [inline src/components/NavBar/NavBar.jsx]: all of this goes back to my comments about changing `handleDeleteClient` to just `handleDelete`, same concept.

just changing it all to be less specific.

File: src/components/NavBar/NavBar.jsx [hunk 4]
```diff
@@ -38,7 +38,7 @@ const NavBar = () => {
   const [openMenu, setOpenMenu] = useState(false);
   const menuId = 'primary-search-account-menu';
 
-  // state for LogoutModal component
+  // state for Modal component
```
Comment (xscottxbrownx) [inline src/components/NavBar/NavBar.jsx]: As a dev, if we just change this - again, I'd be asking "what's this modal for?" Then have to dig around the code and figure it out.  The only way to figure it out is looking at the import statement up top because I have the file name for the component as `LogoutModal`, but if it wasn't specific then we'd have to go to that file and read the code to figure it out.

Eliminating 5 characters here from the code base actually can result in a lot of dev wasted time (when you think about all the new volunteers that come in and out of this project.) Not a good tradeoff for a very minimal amount of "extra" code.

File: src/components/NavBar/NavBar.jsx [hunk 5]
```diff
@@ -55,7 +55,7 @@ const NavBar = () => {
   };
 
   // Event handler for logging out of SOLID POD and removing items from localStorage
-  const handleLogout = () => {
+  const handle = () => {
```
Comment (xscottxbrownx) [inline src/components/NavBar/NavBar.jsx]: same goes here. There's no good benefit in deleting "Logout" here.   Now, "what is it handling? I don't know, I have to go dig around to find out." 

and if you DID think this was a good idea, why only do it with this modal and it's handle function?  There are a lot more handle functions in this file, but you left them specific.

File: src/components/NavBar/NavBar.jsx [hunk 6]
```diff
@@ -143,10 +143,10 @@ const NavBar = () => {
                 />
               )}
               {/* modal/popup renders when showConfirmationModal state is true */}
-              <LogoutModal
```
Comment (xscottxbrownx) [inline src/components/NavBar/NavBar.jsx]: perfect example of the point...

If the name is now just `Modal` and the function is now just `handle`   -  "what IS this modal and what does it do?"  and you'd have to dig around to find out.

File: src/components/NavBar/NavBar.jsx [hunk 7]
```diff
@@ -16,7 +16,7 @@ import Toolbar from '@mui/material/Toolbar';
 import Typography from '@mui/material/Typography';
```
Comment (xscottxbrownx) [inline src/components/NavBar/NavBar.jsx]: There shouldn't be any changes to this file in this PR that is scoped to solve the issue with the delete client modal. Has nothing to do with the Issue/PR at hand.

File: src/components/NavBar/NavBar.jsx (hunk 6 above)
Comment (xscottxbrownx) [reply src/components/NavBar/NavBar.jsx]: and as another point, you didn't do this in `ClientListTableRow`. There you have the component named `<DeleteModal>`, codebase should be consistent one way or the other - but it's ideal to be specific.

File: src/components/DeleteModal/DeleteModal.jsx [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+      <Button
+        variant="outlined"
+        color="error"
+        endIcon={<ClearIcon />}
+        onClick={() => setShowDeleteModal(false)}
+      >
+        NO
+      </Button>
+
+        <Button
```
Comment (xscottxbrownx) [inline src/components/DeleteModal/DeleteModal.jsx]: Just a small formatting issue, this 2nd button should be left one more tab so it lines up with the 1st button code.

We do have a Github Actions check setup for formatting/linting.  Most likely you will need to run the command `npm run prettier:run` in your terminal to format this code correctly to pass the Github Actions testing.

File: src/components/DeleteModal/DeleteModal.jsx [hunk 9]
```diff
@@ -0,0 +1,55 @@
+import React from 'react'
```
Comment (xscottxbrownx) [inline src/components/DeleteModal/DeleteModal.jsx]: Personally I would prefer for this file and component to be named `DeleteClientModal` as it's not re-usable in any other way.  It only does one thing, and that one thing is deleting a client from the client list.

File: src/components/Clients/ClientListTableRow.jsx [hunk 10]
```diff
@@ -103,10 +99,13 @@ const ClientListTableRow = ({ labelId, client, state, dispatch }) => {
...
       </StyledTableCell>
       <StyledTableCell align="center">
-        <IconButton size="large" edge="end" onClick={() => handleDeleteClient()}>
+        <IconButton size="large" edge="end" onClick={() => setShowDeleteModal(true)}>
           <DeleteOutlineOutlinedIcon />
         </IconButton>
       </StyledTableCell>
+
+      {/* modal/popup renders when showDeleteModal state is true */}
+      <DeleteModal showDeleteModal={showDeleteModal} setShowDeleteModal={setShowDeleteModal} client={client} handleDelete={handleDeleteClient} />
```
Comment (xscottxbrownx) [inline src/components/Clients/ClientListTableRow.jsx]: I would personally prefer this component and function to be named `DeleteClientModal` and `handleDeleteClient` - it only does one thing, which is deleting a client from the clients list.    

Naming it `DeleteModal` **1)** seems that it may be re-usable for other purposes. **2)** makes it unknown what it deletes without further exploratory time in the codebase.

File: src/components/Clients/ClientListTableRow.jsx [hunk 11]
```diff
@@ -48,13 +50,7 @@ const ClientListTableRow = ({ labelId, client, state, dispatch }) => {
 
   // Event handler for deleting client from client list
   const handleDeleteClient = async () => {
```
Comment (xscottxbrownx) [inline src/components/Clients/ClientListTableRow.jsx:53]: This function is called from the new `DeleteModal` when a user clicks "Yes", but it doesn't close the modal.  I believe we need to add another line at the end of the function:
`setShowDeleteModal(false);`

**Review by timbot1789 — CHANGES_REQUESTED** [review]
There's something wrong with the git branching here. I think you forked this repo instead of cloning it. Essentially, git thinks you're trying to add in an entirely new project, instead of a part of an existing one. @ajay2827 @xscottxbrownx if you are having trouble fixing it, ping me on discord and I can help you out.

File: src/components/Clients/ClientList.jsx (hunk 1 above)
Comment (ajay2827) [reply src/components/Clients/ClientList.jsx:38]: Now I have added the line, by mistake I delete this line and I haven't  see this

File: src/components/DeleteModal/DeleteModal.jsx (hunk 2 above)
Comment (ajay2827) [reply src/components/DeleteModal/DeleteModal.jsx]: I will Change or name the function and file to be more specific

File: src/components/NavBar/NavBar.jsx (hunk 4 above)
Comment (ajay2827) [reply src/components/NavBar/NavBar.jsx]: I really don't know, may it occur due because of cloning it instead of a fork,

**Comment by ajay2827** [comment]
> There's something wrong with the git branching here. I think you forked this repo instead of cloning it. Essentially, git thinks you're trying to add in an entirely new project, instead of a part of an existing one. @ajay2827 @xscottxbrownx if you are having trouble fixing it, ping me on discord and I can help you out.

It's my mistake instead of forking I have clone it,  Should I first fork it and then create a new Pull Request or there is some other way to fix  this?

File: src/components/Clients/ClientListTableRow.jsx (hunk 11 above)
Comment (ajay2827) [reply src/components/Clients/ClientListTableRow.jsx:53]: yes I should add a line at the end of the function setShowDeleteModal(false)

File: src/components/Clients/ClientListTableRow.jsx (hunk 10 above)
Comment (ajay2827) [reply src/components/Clients/ClientListTableRow.jsx]: Now i rename all function or file , so that it may be more specific

File: src/components/Clients/ClientList.jsx (hunk 1 above)
Comment (leekahung) [reply src/components/Clients/ClientList.jsx:38]: Think we can fix the formatting of this with the spaces so it wouldn't need to appear for files that needs reviewing

File: src/components/NavBar/NavBar.jsx [hunk 12]
```diff
@@ -38,7 +38,7 @@ const NavBar = () => {
   const [openMenu, setOpenMenu] = useState(false);
   const menuId = 'primary-search-account-menu';
 
-  // state for LogoutModal component
+  // state for Logout Modal component
```
Comment (leekahung) [inline src/components/NavBar/NavBar.jsx:41]: Don't think we need to change this comment. The name for the component is LogoutModal, after all.

Unless you would like to rephrase it to `state for the logout modal`

**Review by leekahung — COMMENTED** [review]
Would like to review more after seeing the branch locally on my machine. Is this a fork? I don't see the branch on the PASS repository.

File: src/components/DeleteClientModal/DeleteClientModal.jsx [hunk 13]
```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import DialogContent from '@mui/material/DialogContent';
+import DialogContentText from '@mui/material/DialogContentText';
+import DialogTitle from '@mui/material/DialogTitle';
+import Button from '@mui/material/Button';
+import ClearIcon from '@mui/icons-material/Clear';
+import CheckIcon from '@mui/icons-material/Check';
+
```
Comment (andycwilliams) [inline src/components/DeleteClientModal/DeleteClientModal.jsx:10]: Be sure to add JSDocs here underneath the imports. You can check out the other components to see how that's done.

**Comment by andycwilliams** [comment]
Hello, thank you for your efforts to help.

Could you let us know if you've gone through the README and contribution guidelines yet?

**Comment by xscottxbrownx** [comment]
> Would like to review more after seeing the branch locally on my machine. **Is this a fork?** I don't see the branch on the PASS repository.

@ajay2827 this is definitely a fork of the PASS project, see how it has your username listed before your branch name?
Screenshot 2023-06-17 at 10 19 55 AM

So it's saying you want to merge your branch from *your* repo to the development branch on the *code for pdx* repo, instead of saying you want to merge this branch from the code for pdx repo to the development branch from same repo.

---

Normally it would show like this:
Screenshot 2023-06-17 at 10 22 36 AM

Saying merging a branch to the development branch from same repo

---

So, kind of a pain for you - but we'll need you to clone the project and make the changes again and PR *that* instead.

Before doing anything, read this quick article as well (https://www.theserverside.com/answer/Git-fork-vs-clone-Whats-the-difference). It explains what's happening here pretty well.  

Tim mentioned this before with:
> Essentially, git thinks you're trying to add in an entirely new project, instead of a part of an existing one.

The bottom line is we all want to continue working on this project the way it is, and that is the purpose of `clone`.  `Fork` is usually used if you want to take the project in a new direction on your own - but this project will still exist in it's current direction.

**Comment by timbot1789** [comment]
@xscottxbrownx it shouldnt be that hard of a fix. And @ajay2827 you shouldn't need to redo any work. You can just create a separate branch off the main repo, merge your fork into _that_ branch yourself, and then put up a PR of that branch.

Honestly, we could probably review the code as is. There's nothing wrong with merging in forks. I just want to make sure ajay has his dev environment and flow set up properly.

File: src/components/Clients/ClientListTableRow.jsx [hunk 14]
```diff
@@ -48,16 +51,11 @@ const ClientListTableRow = ({ labelId, client, state, dispatch }) => {
...
-        `You're about to delete ${client.person} from your client list, do you wish to continue?`
-      )
-    ) {
-      return;
-    }
+   
     runNotification(`Deleting "${client.person}" from client list...`, 3, state, dispatch);
     await removeUser(client);
     runNotification(`"${client.person}" deleted from client list...`, 3, state, dispatch);
+    setShowClientDeleteModal(false);
```
Comment (leekahung) [inline src/components/Clients/ClientListTableRow.jsx:58]: Overall, I think this works great. Functionally, it works as intended as well as a confirmation modal.

https://github.com/codeforpdx/PASS/assets/14917816/0a743977-d5a0-4332-bc7f-f43c724b19b5

UI-wise, think we should move the the set function (setShowClientDeleteModal) to the top and perhaps with time of 7s instead of 3s for the first runNotification function in Line 55. That way we don't have to be stuck in the modal until it finishes removeUser and the user could do other things while it's being removed. See below.

https://github.com/codeforpdx/PASS/assets/14917816/bd3c0b0a-9a46-447b-bfd4-46822c5c15e6

File: src/components/Clients/ClientList.jsx [hunk 15]
```diff
@@ -35,7 +35,7 @@ const ClientList = () => {
...
+        Add clients to your list
```
Comment (leekahung) [inline src/components/Clients/ClientList.jsx:38]: Think we could add a space before "Add clients to your list" so that it'll be the same as before. That way it won't pop up as one of the files changed. Think running `npm run prettier:run` should do the trick not just for this file, but the rest of the files which might need formatting.

File: src/components/Clients/ClientListTableRow.jsx [hunk 16]
```diff
@@ -31,6 +32,8 @@ const ClientListTableRow = ({ labelId, client, state, dispatch }) => {
   const { removeUser } = useContext(UserListContext);
   const [pinned, setPinned] = useState(false);
   const [selected, setSelected] = useState(false);
+  
+  const [showDeleteClientModal,setShowClientDeleteModal]=useState(false)
```
Comment (leekahung) [inline src/components/Clients/ClientListTableRow.jsx:36]: Formatting should be fixed after running `npm run prettier:run`.

File: src/components/Clients/ClientListTableRow.jsx [hunk 17]
```diff
@@ -103,10 +101,13 @@ const ClientListTableRow = ({ labelId, client, state, dispatch }) => {
...
       </StyledTableCell>
       <StyledTableCell align="center">
-        <IconButton size="large" edge="end" onClick={() => handleDeleteClient()}>
+        <IconButton size="large" edge="end" onClick={() => setShowClientDeleteModal(true)}>
           <DeleteOutlineOutlinedIcon />
         </IconButton>
       </StyledTableCell>
+
+      {/* modal/popup renders when showDeleteModal state is true */}
+      <DeleteClientModal showDeleteClientModal={showDeleteClientModal} setShowClientDeleteModal={setShowClientDeleteModal} client={client} handleDeleteClient={handleDeleteClient} />
```
Comment (leekahung) [inline src/components/Clients/ClientListTableRow.jsx:110]: Formatting would be fixed after running `npm run prettier:fix`

File: src/components/DeleteClientModal/DeleteClientModal.jsx (hunk 13 above)
Comment (leekahung) [reply src/components/DeleteClientModal/DeleteClientModal.jsx:10]: Yeah, we should add some JSDoc to document the new component. Something basic should do, check out the JSDoc in `AddClientModal.jsx` between Line 13 to 18.

**Review by leekahung — CHANGES_REQUESTED** [review]
I've just checked out the branch from your fork for issue-223. Think it looks pretty good overall, there's just a few things I want to touch (see requested changes).

Be mindful of what to fix during the merge conflicts, but it doesn't look too complicated. Otherwise, it should be set, I think.

I believe Tim and the others have already touched on the branch topic, so I'll leave that out here.

**Comment by xscottxbrownx** [comment]
> @xscottxbrownx it shouldnt be that hard of a fix. And @ajay2827 you shouldn't need to redo any work. You can just create a separate branch off the main repo, merge your fork into _that_ branch yourself, and then put up a PR of that branch.
> 
> Honestly, we could probably review the code as is. There's nothing wrong with merging in forks. I just want to make sure ajay has his dev environment and flow set up properly.

True, this is just an extra step ajay will need to do EACH time trying to merge any new work they've done. Rather than redoing it now once.   But up to them.

**Comment by xscottxbrownx** [comment]
@ajay2827 i hope you are not upset by any of the comments here.  We all appreciate your efforts and the code is good with some minimal tweaks.   Are you still planning to continue to work on this?

**Comment by leekahung** [comment]
Is this PR still being worked on at this @ajay2827 ? It looks like it's almost complete from what I see.

**Comment by xscottxbrownx** [comment]
@ajay2827 it's seems it has been a couple weeks since we've heard anything from you.  If we don't hear anything by this weekend we will probably move forward with finishing your improvements to the codebase and merging it in.  We'd love for you to get credit for this work though.  Are you still planning on working on this at all?

**Comment by leekahung** [comment]
Hey @ajay2827 again, thanks for trying to make a contribution to the project.

Unfortunately, I'm not too sure if you wanted to continue due to a lack of commits/responses to the PR. As such, I'll be closing down this issue.
